src/ast/CodeGeneratorEvaluator.ipp

```diff
@@ -13,40 +13,47 @@ static constexpr char INDENT = '\t';
 
 class CodeGeneratorEvaluator
 {
+  bool containsMain = false;
```

Comment:
It'd be good to have some kind of convention to put all the attributes at the start or the end of a class, and keep it like that for the whole project. I see you put it at the start, I put it at the end in scopeStack, default. Please let's separate the attributes from the rest with private: (even if redundant) to make it clear

Also, more of a fun fact than a comment, but I usually use `attribute` (no underscore) for data (structs) and `_attribute` in classes that have logic. This is just a habbit I have, I guess it came from python with visibility. For me, underscore does not mean "private", but it means it's managed by the class logic, which is a thin nuance

It also helps a lot to see what's an attribute and what's a local scope variable with the blink of an eye. If there is no underscore, it means it's meant to be accessed from the outside (mystruct.myattr), so there is no doubt. If there is an underscore you save yourself the trouble of writing this-> if you're next level lazy